### ISSUE TITLE
Empty lines in where-clause

### ISSUE THREAD

**Issue by stanislav-tkach** [rust-lang/rustfmt#5655]
I would expect the empty line between the `T: std::fmt::Debug` and `T: std::fmt::Display` lines to be removed by rustfmt, but it isn't:
```rust
fn foo<T>(_: T)
where
    T: std::fmt::Debug,

    T: std::fmt::Display,
{
}
```
At least not by the `rustfmt 1.5.1-stable (69f9c33d 2022-12-12)` version.

I'm sorry if it is a duplicate, but I haven't been able to find any related issues.

**Comment by ytmimi** [comment]
Thanks for reaching out! hmm odd behavior, but confirming I can reproduce this!

The [style guide section on where clauses](https://github.com/rust-lang/fmt-rfcs/blob/master/guide/items.md#where-clauses) doesn't mention anything about preserving newlines, so this might just be rustfmt wanting to leave the bounds as the developer wrote them.

I did some digging and it turns out that the newlines are kept because we set `preserve_newline(true)` when building the `ListFormatting` in `rewrite_bounds_on_where_clause`:

https://github.com/rust-lang/rustfmt/blob/ee2bed96d60fd7e46b1fb868f6a8f27e3a8058d0/src/items.rs#L2886-L2889

We can't change the current default behavior because that would be a breaking formatting change, but we could introduce a new, nightly, configuration option to allow you to configure whether or not to preserve newlines between where clause bounds.
